src/pages/SearchPage.js

```diff
@@ -78,8 +83,18 @@ class SearchPage extends Component {
         Timing.end(CONST.TIMING.SEARCH_RENDER);
     }
 
+    componentDidUpdate(prevProps) {
+        if (this.props.reports === prevProps.reports) {
+            return;
+        }
+        this.updateOptions();
+    }
```

Comment:
NAB, but this is kind of weird - I'd expect the options to update when the reports change - but not have to explicitly update them like this.

Reply:
Yeah this is probably outside the scope of the proposed changes so I will resolve for now, but we could do one of these instead:

- introduce more control over when the component renders and get the options when it renders (instead of setting them to state)
- get the options on render, but memoize the method getting the options so that it is more performant (without applying any specific logic about when it should happen)

Reply:
> introduce more control over when the component renders and get the options when it renders (instead of setting them to state)

Considered this.. but was concerned about performance if we recalculated the options on every render.. but maybe it would be just fine.

> get the options on render, but memoize the method getting the options so that it is more performant (without applying any specific logic about when it should happen)

yes! also looked for memoization.. I'm familiarized with `useMemo` when using react with hooks, but searching for memoization in react with classes lead me to use `componentDidUpdate` or external libraries like `memoizeOne`... and now I see that `memoizeOne` is already in our project so I can give that a try

I'll test removing the options from the state and use `memoizeOne`

Reply:
let's save that for another PR please 🙇